app/controllers/sitemap_controller.rb

```diff
@@ -0,0 +1,55 @@
+class SitemapController < ActionController::Metal
+
+  include AbstractController::Rendering
+  include ActionController::MimeResponds
+  include ActionController::DataStreaming
+  include ActionController::RackDelegation
+  include ActionController::Rescue
+  include ActionController::Head
+
+  def sitemap
+    render_site_map
+  end
+
+  private
+
+    def render_site_map
+      community = request.env[:current_marketplace]
+      if community.deleted?
+        head :not_found
+        return
+      end
+      if community.private?
+        head :forbidden
+        return
+      end
+      respond_to do |format|
+
+        data = Rails.cache.fetch(community.id, expires_in: 24.hours) do
+          adapter = SitemapGenerator::NeverWriteAdapter.new
+
+          community_id = community.id
+
+          SitemapGenerator::Sitemap.create(
+                :default_host => request.base_url,
+                :verbose => false,
+                :adapter => adapter) do
+                  SitemapController.find_open_listings(community_id).each do |l|
+                    add listing_path(id: l.to_param), :lastmod => l.updated_at
+                  end
+              end
+
+          #SitemapGenerator::Interpreter.run(:default_host => "http://www.bar.com", :adapter => adapter)
+          data = adapter.get_data
+        end
+        format.xml_gz { send_data  ActiveSupport::Gzip.compress(data) }
+        format.html { send_data  ActiveSupport::Gzip.compress(data) }
+      end
+    end
+
+    def self.find_open_listings(community_id)
+      listings = Listing.currently_open.where(community_id: community_id).limit(SitemapGenerator::MAX_SITEMAP_LINKS)
```

Comment:
It's easy enough to go down from that number and make that a configuration parameter.  If we go above that number, we need to split apart the sitemaps (which isn't supported with the current code).  How many marketplaces do you have that have more than 50k listings?

That said, if you're creating a sitemap for SEO, I can't imagine wanting to have fewer than every listing available for google, bing, etc to index.  What am I missing?